### PR TITLE
Hybrid support for MorphToMany relationship

### PR DESCRIPTION
**Pull request by hans-thomas**
Hello everybody, this PR adds hybrid support for the MorphToMany relationship.
I tested it in two scenarios. First, an SQL model makes a MorphToMany relation to a MongoDB model. In the second one, A MongoDB Model makes a MorphToMany to an SQL model.

### CODE REVIEW

**Review by GromNaN — APPROVED** [review]

**Comment by GromNaN** [comment]
Can you update this PR please. There is some conflicts.

File: src/Relations/MorphToMany.php [hunk 1]
```diff
@@ -185,15 +213,19 @@ public function attach($id, array $attributes = [], $touch = true)
...
+                            $this->morphType => $model->getMorphClass(),
+                        ],
+                    ], true);
+                } else {
+                    $this->addIdToParentRelationData($id);
+                }
 
                 // Attach the new parent id to the related model.
-                $model->push($this->foreignPivotKey, $this->parseIds($this->parent), true);
+                $model->push($this->foreignPivotKey, (array) $this->parent->{$this->parentKey}, true);
```
Comment (GromNaN) [inline src/Relations/MorphToMany.php:228]: `(array)` cast should not be used. If the value is an `ObjectId`, the result is not the one you expect.

```php
var_dump((array) new MongoDB\BSON\ObjectId());
/*
array(1) {
  'oid' =>
  string(24) "6577071f3fded0903a01f030"
}
*/
```

Here we know it's not a list of ids, so a simple array wrapping is fine.
```suggestion
                $model->push($this->foreignPivotKey, [$this->parent->{$this->parentKey}], true);
```

**Review by GromNaN — COMMENTED** [review]
Thank you for the update.

**Review by GromNaN — APPROVED** [review]

**Comment by GromNaN** [comment]
Thank you @hans-thomas

File: src/Relations/MorphToMany.php (hunk 1 above)
Comment (hans-thomas) [reply src/Relations/MorphToMany.php:228]: I will double-check it later. thanks for the review❤️

Comment (hans-thomas) [reply src/Relations/MorphToMany.php:228]: The `ObjectId` fields are always `string`. I tested this with a primary key and a custom key and a custom key was cast to `ObjectId`.
In all cases, the `$this->parent->{$this->parentKey}` value is `string`. Your suggestion is more readable, but I don't see any bug here to open a PR.

Comment (GromNaN) [reply src/Relations/MorphToMany.php:228]: Yes, that's something we'll have to fix if we manage to store references as ObjectId. But there are many other hurdles to overcome, and I think this notion of "all keys are strings" is too hard-wired into Eloquent anyway.

**Pull request merged**